### DISCUSSION TITLE
Map the inner value of a resource

### DISCUSSION THREAD

**Discussion by jonahlund** [leptos-rs/leptos#2034] [Q&A] [answered]
Is it possible to map the inner value of a resource to another value? This could be beneficial in a few cases, like for example say i have a `UserContext` which wraps a `Resource<(), Option<User>>`, which would allow any component to access the resource and its various features, but this approach would mean that `User` has to be cloned which is not ideal, it would be more ideal to wrap `Resource<(), Option<Rc<User>>>` for the context to avoid cloning the user, but the resource can't return `Rc` since its not serializable, which is where it would be beneficial to map the resource value to another value

**Comment by gbj** [comment, marked as answer]
This seems to suggest something like `Signal::derive(|| resource.map(|user| /* do something */))` would be appropriate for your use case.

I'd also note though that it's almost certainly the case that cloning `User` is not going to be significant relative to the overall cost of doing things like rendering your UI in the DOM. (I'm assuming `User` is something like a handful of string fields)

Also, `Rc` is serializable, you just need to have the `rc` feature enabled on `serde`.
